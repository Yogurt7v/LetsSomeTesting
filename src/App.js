import './App.css';
import { useEffect, useState } from 'react';
import Users from './users/users';

function App() {
  const [data, setData] = useState(null);
  const [toggle, setToggle] = useState(false);
  const [value, setValue] = useState('');

  const toggleFunc = () => {
    setToggle((prev) => !prev);
  };

  useEffect(() => {
    const setNewData = () => {
      setTimeout(() => {
        setData('Hello World2');
      }, 1000);
    };

    setNewData();

    return () => {
      clearTimeout();
    };
  }, []);

  return (
    <div>
      {data && <div data-testid="toggle">check</div>}
      <h1>Hello World</h1>

      {value.length < 3 ? (
        <p data-testid="error">Too short</p>
      ) : (
        <h3 data-testid="h3">{value}</h3>
      )}
      <input
        type="text"
        placeholder="Type here..."
        value={value}
        onChange={(e) => setValue(e.target.value)}
      ></input>
      <button onClick={toggleFunc}>Push me</button>
      {toggle && <div>true</div>}
      <Users />
    </div>
  );
}

export default App;
