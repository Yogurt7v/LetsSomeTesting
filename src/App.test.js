import { prettyDOM, render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';

describe('проверка функции App', () => {
  test.skip('renders learn react link', () => {
    render(<App />);
    const helloWorldElement = screen.getByText(/hello world/i);
    const buttonElement = screen.getByRole('button');
    const inputElement = screen.getByPlaceholderText(/Type here/i);
    // console.log(prettyDOM(inputElement));

    // expect(helloWorldElement).toBeInTheDocument();
    // expect(buttonElement).toBeInTheDocument();
    // expect(inputElement).toBeInTheDocument();
    // expect(inputElement).toMatchSnapshot();
  });

  test('проверка ввода', async () => {
    render(<App />);
    const inputElement = screen.getByPlaceholderText(/Type here/i);
    expect(screen.queryByTestId('h3')).toBeNull();
    // fireEvent.input(inputElement, { target: { value: 'test' } });
    userEvent.type(inputElement, 'test');
    expect(await screen.findByTestId('h3')).toHaveTextContent('test');
  });
});
