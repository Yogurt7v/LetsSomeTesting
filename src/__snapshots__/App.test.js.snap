// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка функции App renders learn react link 1`] = `
<input
  placeholder="Type here..."
  type="text"
/>
`;
